XenModel/Actions/Network/CreateSriovAction.cs

```diff
@@ -73,8 +73,11 @@ protected override void Run()
             Connection.ExpectDisruption = true;
 
             // Enable SR-IOV network on Pool Master
-            RelatedTask = Network_sriov.async_create(Session, pifOnMaster.opaque_ref, networkRef);
-            PollToCompletion(0, 100);
+            if(pifOnMaster != null)
+            {
+                RelatedTask = Network_sriov.async_create(Session, pifOnMaster.opaque_ref, networkRef);
+                PollToCompletion(0, 100);
+            }

```

Comment:
We should move the line `selectedPifs.Remove(pifOnMaster);` inside the `if` block as well, to avoid an unnecessary operation (which is an O(n) operation).

Also (and apologies for not noticing it earlier), in the block above we do an `async_create` for the master while for the slaves we do `create`. We should be consistent. 
The `PollToCompletion(0,100)` call above will set the action's progress to 100%, while that  shouldn't happen, the action should be 100% completed only when all SR-IOV networks on the slaves are created.